apollo-router/src/plugins/external.rs

```diff
@@ -326,63 +475,271 @@ impl Plugin for ExternalPlugin {
+
+                    let payload = Externalizable {
+                        version: EXTERNALIZABLE_VERSION,
+                        stage: PipelineStep::SubgraphResponse.to_string(),
+                        control: None,
+                        id: TraceId::maybe_new().map(|id| id.to_string()),
+                        headers: headers_to_send,
+                        body: body_to_send,
+                        context: context_to_send,
+                        sdl: None,
+                        uri: None,
+                        service_name,
+                    };
+
+                    tracing::debug!(?payload, "externalized output");
+                    response.context.enter_active_request().await;
+                    let co_processor_result = payload.call(http_client, &coprocessor_url).await;
+                    response.context.leave_active_request().await;
+                    tracing::debug!(?co_processor_result, "co-processor returned");
+                    let co_processor_output = co_processor_result?;
+
+                    validate_coprocessor_output(
+                        &co_processor_output,
+                        PipelineStep::SubgraphResponse,
+                    )?;
+
+                    // Third, process our reply and act on the contents. Our processing logic is
+                    // that we replace "bits" of our incoming response with the updated bits if they
+                    // are present in our co_processor_output. If they aren't present, just use the
+                    // bits that we sent to the co_processor.
+
+                    let new_body: crate::graphql::Response = match co_processor_output.body {
+                        Some(value) => serde_json::from_value(value)?,
+                        None => body,
+                    };
+
+                    response.response = http::Response::from_parts(parts, new_body);
+
+                    if let Some(context) = co_processor_output.context {
+                        response.context = context;
+                    }
+
+                    if let Some(headers) = co_processor_output.headers {
+                        *response.response.headers_mut() = internalize_header_map(headers)?;
+                    }
+
+                    Ok::<subgraph::Response, BoxError>(response)
+                }
+            })
+        });
+
+        fn external_service_span() -> impl Fn(&subgraph::Request) -> tracing::Span + Clone {
+            move |_request: &subgraph::Request| {
+                tracing::info_span!(
+                    EXTERNAL_SPAN_NAME,
+                    "external service" = stringify!(subgraph::Request),
+                    "otel.kind" = "INTERNAL"
+                )
+            }
         }
+
+        ServiceBuilder::new()
+            .instrument(external_service_span())
+            .option_layer(request_layer)
+            .option_layer(response_layer)
+            .buffer(20_000)
```

Comment:
We had a `buffered` method somewhere in a trait to not use hardcoded value for buffer. If it's still existing we should use it

Reply:
it still exists, but with @garypen's PR on `AsyncCheckPoint` it should not be necessary anymore

Reply:
Ok let's have a followup pass at it once it lands.